### PR TITLE
Update repo groups for kubernetes

### PR DESCRIPTION
**Pull request by spiffxp**
This address https://github.com/cncf/devstats/issues/158 and supercedes https://github.com/cncf/devstats/pull/167

This also handles https://github.com/cncf/devstats/issues/173 by keeping old and new repo names around (unless there is a better way to tell devstats how to keep track of repo renames and migrations?

This is based on the script in https://github.com/kubernetes/community/pull/3853

### CODE REVIEW

File: scripts/kubernetes/repo_groups.sql [hunk 1]
```diff
@@ -38,20 +45,21 @@ update gha_repos set repo_group = 'SIG API Machinery' where name in (
 
 update gha_repos set repo_group = 'SIG Apps' where name in (
   'kubernetes-sigs/application',
+  'kubernetes-sigs/execution-hook',
   'kubernetes/examples',
   'kubernetes/kompose'
 );
 
 update gha_repos set repo_group = 'SIG Architecture' where name in (
-  'kubernetes-incubator/spartakus',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:46]: this migrated to steering committee

File: scripts/kubernetes/repo_groups.sql [hunk 2]
```diff
@@ -38,20 +45,21 @@ update gha_repos set repo_group = 'SIG API Machinery' where name in (
...
-  'kubernetes-sigs/architecture-tracking',
   'kubernetes/api',
   'kubernetes/contrib',
-  'kubernetes/kubernetes-template-project',
-  'kubernetes/steering',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:51]: these two migrated to steering committee

File: scripts/kubernetes/repo_groups.sql [hunk 3]
```diff
@@ -38,20 +45,21 @@ update gha_repos set repo_group = 'SIG API Machinery' where name in (
...
 update gha_repos set repo_group = 'SIG Apps' where name in (
   'kubernetes-sigs/application',
+  'kubernetes-sigs/execution-hook',
   'kubernetes/examples',
   'kubernetes/kompose'
 );
 
 update gha_repos set repo_group = 'SIG Architecture' where name in (
-  'kubernetes-incubator/spartakus',
-  'kubernetes-sigs/architecture-tracking',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:47]: this was archived and didn't have substantive activity https://github.com/kubernetes/org/issues/738

File: scripts/kubernetes/repo_groups.sql [hunk 4]
```diff
@@ -110,9 +140,6 @@ update gha_repos set repo_group = 'SIG Contributor Experience' where name in (
 
 update gha_repos set repo_group = 'SIG Docs' where name in (
   'kubernetes-incubator/reference-docs',
-  'kubernetes/kubernetes-docs-ja',
-  'kubernetes/kubernetes-docs-ko',
-  'kubernetes/kubernetes-docs-zh',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:115]: The localization repos were archived, and most of their activity went into kubernetes/website: https://github.com/kubernetes/org/issues/154

File: scripts/kubernetes/repo_groups.sql [hunk 5]
```diff
@@ -141,23 +175,29 @@ update gha_repos set repo_group = 'SIG Network' where name in (
   'kubernetes-incubator/ip-masq-agent',
   'kubernetes/dns',
   'kubernetes/ingress-gce',
-  'kubernetes/ingress-nginx',
-  'kubernetes/pkg'
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:145]: I'm honestly not sure what this was and couldn't find kubernetes/org history for it; it now 404's

File: scripts/kubernetes/repo_groups.sql [hunk 6]
```diff
@@ -174,48 +214,73 @@ update gha_repos set repo_group = 'SIG Scheduling' where name in (
...
 
 update gha_repos set repo_group = 'SIG Service Catalog' where name in (
-  'kubernetes-incubator/service-catalog'
+  'kubernetes-incubator/service-catalog',
+  'kubernetes-sigs/minibroker',
+  'kubernetes-sigs/service-catalog'
 );
 
 update gha_repos set repo_group = 'SIG Storage' where name in (
-  'kubernetes-csi/csi-driver-cinder',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:181]: This was unused/removed https://github.com/kubernetes/org/issues/317

File: scripts/kubernetes/repo_groups.sql [hunk 7]
```diff
@@ -174,48 +214,73 @@ update gha_repos set repo_group = 'SIG Scheduling' where name in (
...
+  'kubernetes-csi/cluster-driver-registrar',
   'kubernetes-csi/csi-driver-flex',
   'kubernetes-csi/csi-driver-host-path',
+  'kubernetes-csi/csi-driver-image-populator',
   'kubernetes-csi/csi-driver-iscsi',
   'kubernetes-csi/csi-driver-nfs',
-  'kubernetes-csi/csi-lib-common',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:186]: This was unused/removed https://github.com/kubernetes/org/issues/718

File: scripts/kubernetes/repo_groups.sql [hunk 8]
```diff
@@ -174,48 +214,73 @@ update gha_repos set repo_group = 'SIG Scheduling' where name in (
...
+  'kubernetes-csi/csi-release-tools',
   'kubernetes-csi/csi-test',
   'kubernetes-csi/docs',
   'kubernetes-csi/driver-registrar',
   'kubernetes-csi/drivers',
   'kubernetes-csi/external-attacher',
   'kubernetes-csi/external-provisioner',
+  'kubernetes-csi/external-resizer',
   'kubernetes-csi/external-snapshotter',
-  'kubernetes-csi/kubernetes-csi-migration-library',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:196]: This was deleted https://github.com/kubernetes/org/issues/718

File: scripts/kubernetes/repo_groups.sql [hunk 9]
```diff
@@ -174,48 +214,73 @@ update gha_repos set repo_group = 'SIG Scheduling' where name in (
...
+  'kubernetes-sigs/sig-storage-local-static-provisioner',
   'kubernetes/csi-api',
+  'kubernetes/csi-translation-lib',
   'kubernetes/git-sync'
 );
 
 update gha_repos set repo_group = 'SIG Testing' where name in (
   'kubernetes-sigs/kind',
   'kubernetes-sigs/testing_frameworks',
-  'kubernetes/publishing-bot',
```
Comment (spiffxp) [inline scripts/kubernetes/repo_groups.sql:209]: This migrated to sig-release

**Review by spiffxp — COMMENTED** [review]
Added context for the deletions.  Anything I didn't annotate was sorting or shuffling of a trailing `,`

**Comment by dankohn** [comment]
Thank you, @spiffxp. This is great work. How often does your hack need to be rerun?

**Comment by spiffxp** [comment]
We could get to a point where we update the file every time we update our repos.  I'm not sure that's necessary.  I was envisioning something more like once per quarter, and am open to feedback.

**Comment by lukaszgryglicki** [comment]
This requires regenerating all dashboards data so is a bit time-consuming. Once per quarter is of course OK.

**Pull request merged**